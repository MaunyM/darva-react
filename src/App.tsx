import React, { useState, useEffect } from 'react';
import './App.css';
import TodoList from './components/TodoList';
import Item from './type/Item';
import TodoContext, { ITodoContext } from './context/TodoContext';
import TodoForm from './components/TodoForm';
import { BrowserRouter as Router, Switch, Route, NavLink } from 'react-router-dom';
import TodoItem from './components/TodoItem';
import TodoItemById from './components/TodoItemById';

async function fetchItems(): Promise<Item[]> {
  const response = await fetch('https://api.larus.fr/pwa/post')
  return response.json()
}

function postItem(item: Item): Promise<any> {
  return fetch('https://api.larus.fr/pwa/post', {
    method: 'POST',
    body: JSON.stringify(item)
  })
}

function App() {

  const [todos, setTodos] = useState<Item[]>([])

  useEffect(() => {
    const truande = async () => {
      try {
        const items = await fetchItems()
        setTodos(items)
      } catch {

      }
    }
    truande()
  }, [])

  useEffect(() => {
    fetchItems().then(items => setTodos(items))
  }, [])

  const itemIsDone = (item: Item) => {
    const nouvelleListe = todos.map((itemCourant) => {
      if (item.name === itemCourant.name) {
        return { ...item, isDone: true }
      } else {
        return itemCourant
      }
    }
    )
    setTodos(nouvelleListe)
  }

  const itemIsUndone = (item: Item) => {
    const nouvelleListe = todos.map((itemCourant) => {
      if (item.name === itemCourant.name) {
        return { ...item, isDone: false }
      } else {
        return itemCourant
      }
    }
    )
    setTodos(nouvelleListe)
  }

  const addItem = (item: Item) => {
    postItem(item).then(() =>
      setTodos((prevState) => ([...prevState, item]))
    )
  }

  const cleanItems = () => setTodos([])

  const context: ITodoContext = {
    todos,
    addItem,
    cleanItems,
    itemIsDone,
    itemIsUndone
  }
  return (
    <TodoContext.Provider value={context}>
      <div className="App">
        <Router>
          <nav>
            <NavLink exact={true} to={'/'}>Home</NavLink>
            <NavLink to={'/add'}>Ajouter</NavLink>
          </nav>
          <Switch>
            <Route path={'/add'}>
              <TodoForm />
            </Route>
            <Route path={'/item/:id'}>
              <TodoItemById />
            </Route>
            <Route path={'/'}>
              <TodoList />
            </Route>
          </Switch>
        </Router>
      </div>
    </TodoContext.Provider>
  );

}

export default App;
