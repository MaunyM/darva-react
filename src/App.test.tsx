import React from 'react';
import renderer from 'react-test-renderer';
import App from './App';

test('renders learn react link', () => {
  const component = renderer.create(
    <App/>,
  );
  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});