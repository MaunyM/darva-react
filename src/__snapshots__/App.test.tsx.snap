// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders learn react link 1`] = `
<div
  className="App"
>
  <nav>
    <a
      aria-current="page"
      className="active"
      href="/"
      onClick={[Function]}
      style={Object {}}
    >
      Home
    </a>
    <a
      aria-current={null}
      href="/add"
      onClick={[Function]}
    >
      Ajouter
    </a>
  </nav>
  <div
    className="TodoList"
  >
    0
     Ã  faire
    <button
      onClick={[Function]}
    >
      Vider la liste
    </button>
  </div>
</div>
`;
